pdns/pdns_recursor.cc

```diff
@@ -2451,13 +2459,19 @@ void broadcastFunction(const pipefunc_t& func, bool skipSelf)
 
 void distributeAsyncFunction(const string& packet, const pipefunc_t& func)
 {
+  if (t_id != 0) {
```

Comment:
possibly stupid q: why do all of these funcs have different t_id checks?

Reply:
Not stupid at all, there is some level of pain hidden here. `distributeAsyncFunction()` can only be called by the distributor thread (`t_id` 0 at the moment), `broadcastFunction()` only by the new `handler` thread (`t_id` -1), except once at startup where it's called by the first thread (`t_id` 0), and `broadcastAccFunction()` only by the new `handler` thread.